packages/ui/src/views/repo/repo-sidebar/index.tsx

```diff
@@ -1,57 +1,82 @@
 
 interface RepoSidebarProps {
   hasHeader?: boolean
   hasSubHeader?: boolean
+  repoId: string
+  spaceId: string
   selectedBranch: BranchSelectorListItem
   branchList: BranchSelectorListItem[]
   tagList: BranchSelectorListItem[]
   selectBranch: (branch: BranchSelectorListItem) => void
   navigateToNewFile: () => void
+  navigateToNewFolder: () => void
   navigateToFile: (file: string) => void
-  filesList: string[] | undefined
+  filesList?: string[]
+  children: ReactNode
 }
 
 export const RepoSidebar = ({
   hasHeader,
   hasSubHeader,
+  repoId,
+  spaceId,
   selectedBranch,
   branchList,
   tagList,
   selectBranch,
   navigateToNewFile,
+  navigateToNewFolder,
   navigateToFile,
-  filesList
+  filesList,
+  children
 }: RepoSidebarProps) => {
   return (
-    <SandboxLayout.LeftSubPanel hasHeader={hasHeader} hasSubHeader={hasSubHeader}>
-      <SandboxLayout.Content>
-        <div className="flex flex-col gap-5">
-          <div className="grid w-full auto-cols-auto grid-flow-col grid-cols-[1fr] items-center gap-3">
+    <SandboxLayout.LeftSubPanel className="w-[248px]" hasHeader={hasHeader} hasSubHeader={hasSubHeader}>
+      <SandboxLayout.Content className="flex h-full overflow-hidden p-0">
+        <div className="flex w-full flex-col gap-3 p-5 pb-0 pr-0">
+          <div className="grid w-full auto-cols-auto grid-flow-col grid-cols-[1fr] items-center gap-3 pr-5">
             {branchList && (
               <BranchSelector
                 selectedBranch={selectedBranch}
                 branchList={branchList}
                 tagList={tagList}
                 onSelectBranch={selectBranch}
-                // TODO: spaceId and repoId transfer is required
-                repoId="repoId"
-                spaceId="spaceId"
+                repoId={repoId}
+                spaceId={spaceId}
               />
             )}
-            <ButtonGroup.Root
-              spacing="0"
-              className="shadow-border h-full overflow-hidden rounded-md shadow-[inset_0_0_0_1px]"
-            >
-              <Button size="sm" variant="ghost" className="w-8 rounded-none border-l p-0" onClick={navigateToNewFile}>
-                <Icon size={15} name="add-file" className="text-primary/80" />
+            <ButtonGroup.Root spacing="0" className="h-full overflow-hidden rounded shadow-as-border shadow-borders-2">
+              <Button
+                size="icon"
+                variant="ghost"
+                className="rounded-none border-l border-borders-2 p-0"
+                onClick={navigateToNewFolder}
+              >
+                <Icon size={16} name="add-folder" className="text-icons-3" />
+              </Button>
+              <Button
+                size="icon"
+                variant="ghost"
+                className="rounded-none border-l border-borders-2 p-0"
+                onClick={navigateToNewFile}
+              >
+                <Icon size={16} name="add-file" className="text-icons-3" />
```

Comment:
When only including an icon in a clickable element (link or button), we need a `aria-label` to describe what the thing does for a11y
```suggestion
              <Button
                size="icon"
                variant="ghost"
                className="rounded-none border-l border-borders-2 p-0"
                onClick={navigateToNewFolder}
                aria-label="Create new folder"
              >
                <Icon size={16} name="add-folder" className="text-icons-3" />
              </Button>
              <Button
                size="icon"
                variant="ghost"
                className="rounded-none border-l border-borders-2 p-0"
                onClick={navigateToNewFile}
                aria-label="Create new file"
              >
                <Icon size={16} name="add-file" className="text-icons-3" />
```